services/apps/src/apps/service/apps/de/listings.clj

```diff
@@ -125,18 +128,21 @@
   "Retrieves the list of app groups that are visible to all users, the current user's app groups, or
    both, depending on the :public param."
   [user {:keys [public] :as params}]
-  (if (contains? params :public)
-    (if-not public
-      (get-workspace-app-groups user params)
-      (get-visible-app-groups-for-workspace nil user params))
-    (get-visible-app-groups user params)))
+  (let [perms  (iplant-groups/load-app-permissions (:shortUsername user))
+        params (assoc params :app-ids (set (keys perms)))]
+    (if (contains? params :public)
+      (if-not public
+        (get-workspace-app-groups user params)
+        (get-visible-app-groups-for-workspace nil user params))
+      (get-visible-app-groups user params))))
 
 (defn get-admin-app-groups
   "Retrieves the list of app groups that are accessible to administrators. This includes all public
    app groups along with the trash group."
-  [params]
-  (conj (vec (get-app-groups nil params))
-        (format-trash-category nil nil params)))
+  [user params]
+  (let [params (assoc params :admin true)]
```

Comment:
The user has to be passed to iplant-groups in order to look up app permissions, so we need an alternate way to indicate when an admin listing is being requested.